kube/src/api/metadata.rs

```diff
@@ -16,18 +16,23 @@ use std::borrow::Cow;
 ///
 /// This avoids a bunch of the unnecessary unwrap mechanics for apps.
 pub trait Meta {
-    /// Types that know their metadata at compile time should select `Family = ()`.
-    /// Types that require some information at runtime should select `Family`
+    /// Type information for types that do not know their resource information at compile time.
+    ///
+    /// Types that know their metadata at compile time should select `RuntimeType = ()`.
+    /// Types that require some information at runtime should select `RuntimeType`
     /// as type of this information.
-    type Family: Send + Sync + 'static;
+    ///
+    /// See [`DynamicObject`] for a valid implementation of non-k8s-openapi resources.
+    type RuntimeType: Send + Sync + 'static;
```

Comment:
`DynamicType`?
```rust
/// See [`DynamicObject`] for a valid implementation of non-k8s-openapi resources.

/// This function accepts `K::RuntimeType` so it can be used with dynamic resources.
```

C# Dynamic Type
> C# 4 introduces a new type, `dynamic`. The type is a static type, but an object of type `dynamic` bypasses static type checking.
>
> [Using type dynamic - C# Programming Guide | Microsoft Docs](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/types/using-type-dynamic)

---

`TypeId`? `TypeInfo`?

> The `typeid` keyword is used to determine the `class` of an `object` at run time. It returns a reference to `std::type_info` object
>
> [Run-time type information - Wikipedia](https://en.wikipedia.org/wiki/Run-time_type_information#typeid)

Reply:
`DynamicType` is very similar, I am also happy with that.

`TypeInfo` is also great. Nice and short, which is probably smarter in this core trait.
On the other hand it's just a secondary storage type for the type information, and only for dynamic cases. It possibly fails to convey the optionality/dynamic-only nature of it.

Reply:
:+1: for me on `DynamicType`